File: day06/puzzle.rb
# frozen_string_literal: true

def find_packet_marker(input, seq_length)
  input.chars.each_cons(seq_length).with_index do |seq, idx|
    return idx + seq_length if seq.uniq.length == seq_length
  end
end

input = File.read(*ARGV)
p find_packet_marker(input, 4) # p1-> 1912

p find_packet_marker(input, 14) # p2-> 2122

File: day06/puzzle.js
const fs = require("fs");
const read = (path) => {
  return fs.readFileSync(path, { encoding: "utf8" });
}

const input = read(process.argv[2]);

const findPacketMarker = (seqLength) => {
  for (let i = 0; i < input.length; i++) {
    const endIdx = i + seqLength
    const slice = input.slice(i, endIdx);
    if (new Set(slice).size === seqLength) {
      return endIdx;
    }
  }
}

console.log(findPacketMarker(4)); // Part1

console.log(findPacketMarker(14)); // Part2
